plugins/inputs/whois/whois_test.go

```diff
@@ -0,0 +1,198 @@
+package whois
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/likexian/whois"
+	"github.com/stretchr/testify/require"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/parsers/influx"
+	"github.com/influxdata/telegraf/testutil"
+)
+
+// Make sure Whois implements telegraf.Input
+var _ telegraf.Input = &Whois{}
+
+func TestWhoisConfigInitialization(t *testing.T) {
+	tests := []struct {
+		name               string
+		domains            []string
+		server             string
+		IncludeNameServers bool
+		timeout            config.Duration
+		expectErr          bool
+	}{
+		{
+			name:      "Valid Configuration",
+			domains:   []string{"example.com", "google.com"},
+			server:    "whois.example.org",
+			timeout:   config.Duration(10 * time.Second),
+			expectErr: false,
+		},
+		{
+			name:      "No Domains Configured",
+			domains:   nil,
+			timeout:   config.Duration(5 * time.Second),
+			expectErr: true,
+		},
+		{
+			name:      "Invalid Timeout (Zero Value)",
+			domains:   []string{"example.com"},
+			timeout:   config.Duration(0),
+			expectErr: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			plugin := &Whois{
+				Domains: test.domains,
+				Timeout: test.timeout,
+				Server:  test.server,
+				Log:     testutil.Logger{},
+			}
+
+			err := plugin.Init()
+
+			if test.expectErr {
+				require.Error(t, err, "Expected error but got none")
+				return
+			}
+
+			require.NoError(t, err, "Unexpected error during Init()")
+		})
+	}
+}
+
+func TestWhoisCases(t *testing.T) {
+	// Get all directories in testcases
+	folders, err := os.ReadDir("testcases")
+	require.NoError(t, err, "Failed to read testcases directory")
+
+	// Register the WHOIS plugin inside the test
+	inputs.Add("whois", func() telegraf.Input {
+		return &Whois{}
```

Comment:
I would configure a timeout to not block testing if something goes wrong...
```suggestion
		return &Whois{Timeout: config.Duration(3*time.Second)}
```

Reply:
Added.